Cue/app/tabs/MenuProfileItem.js

```diff
@@ -36,20 +36,42 @@ const styles = {
 class MenuProfileItem extends React.Component {
   props: {
     name: string,
+    navigator: Navigator,
     extraStyles?: Object,
-    onLogOut: () => void,
 
     // From Redux:
     logOut: () => void,
+    localChanges: {},
   }
 
   _onMenuAction = (index: number) => {
-    this._logOut()
-  }
-
-  _logOut = () => {
-    LoginManager.logOut()
-    this.props.logOut()
+    if (this.props.localChanges && this.props.localChanges.length) {
+      this.props.syncLibrary(this.props.localChanges).then(failedSyncs =>{
+        if (failedSyncs && failedSyncs.length) {
+          Alert.alert(
+            'Failed to sync changes',
+            'Resolve conflicts or logout and lose local changes',
+            [
+              {text: "Logout", onPress: () => this.props.logOut()},
+              {text: "Resolve", onPress: () => this.props.navigator.push({failedSyncs})}
+            ],
+            { cancelable: false }
+          )
+        } else {
+          this.props.logOut()
+        }
+      }).catch(e => {
+        console.warn('Failed to sync changes', e)
+        Alert.alert(
+          'Could not sync local changes',
+          'Logout and lose local changes?',
+          [{text: 'Logout', onPress: () => this.props.logOut()},
+           {text: 'Cancel', style: 'cancel'}]
```

Comment:
Alternatively, you could add a little extra to better communicate why we can't sync right now and what "local changes" will be lost:

-----
**Some changes haven’t been synced to the Cue cloud yet**

Can’t connect to the Cue cloud right now.

If you log out now, you will lose all the changes you made while offline.

`Cancel` `Log Out Anyway`

-----